llvm/lib/Target/Mips/MipsSEFrameLowering.cpp

```diff
@@ -425,9 +425,9 @@ void MipsSEFrameLowering::emitPrologue(MachineFunction &MF,
   unsigned ADDiu = ABI.GetPtrAddiuOp();
   unsigned AND = ABI.IsN64() ? Mips::AND64 : Mips::AND;
 
-  // C0 is always live in.  In non-CHERI MIPS, C0 is treated as a read-only
-  // register that is implicitly read by loads and stores.
-  MBB.addLiveIn(Mips::DDC);
+  // DDC is always live in.
+  if (STI.isCheri())
+    MBB.addLiveIn(Mips::DDC);
```

Comment:
Not needed now there's no modelling of it, this isn't done for CHERI-RISC-V

Reply:
Fixed in the new version.